tests/system/datastore/test_v3io.py

```diff
@@ -185,8 +201,21 @@ def test_v3io_large_object_put(self):
             f"test_v3io_large_object_put: total time: {time.time() - first_start_time} seconds"
         )
 
-    def test_put_get_and_download(self):
-        data_item, _ = self._get_data_item()
+    @pytest.mark.parametrize("use_secrets_as_parameters", [True, False])
+    def test_put_get_and_download(
+        self, use_datastore_profile, use_secrets_as_parameters
+    ):
+        secrets = {}
+        if use_secrets_as_parameters:
+            os.environ["V3IO_ACCESS_KEY"] = "wrong_token"
```

Comment:
The thing with setting the environment variable like this is that it's sticky across tests, and can cause situations where which tests are run and in what order can affect whether a test will pass or fail. It's better to save the original and reset the variable in a finally clause.

Reply:
I have a fixture that accomplishes this and runs before each test.
It is a better solution than adding a 'finally' statement at the end of each test.

Reply:
please look at :
```

@pytest.fixture(autouse=True)
    def setup_before_each_test(self, use_datastore_profile):
```

Reply:
and anyway, in v3io we inherit from TestMLRunSystem class.
In `setup_method` you overwrite this env variable in `self._setup_env(self._get_env_from_file())`